agent/lib/kontena/websocket_client.rb

```diff
@@ -182,9 +198,9 @@ def handle_invalid_token
       EM.next_tick { abort('Shutting down ...') }
     end
 
-    def handle_invalid_version
+    def handle_invalid_version(reason)
       agent_version = Kontena::Agent::VERSION
-      error "master does not accept our version (#{agent_version}), shutting down ..."
+      error "master does not accept our version: #{reason}"
```

Comment:
Also not backwards-compatible with older server version... they don't send a `close reason: ...`.